File: app/src/main/java/com/hjc/base/ui/senior/view/ViewCanvasActivity.java
package com.hjc.base.ui.senior.view;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.view.View;

import com.alibaba.android.arouter.facade.annotation.Route;
import com.hjc.base.R;
import com.hjc.base.constant.RoutePath;
import com.hjc.base.databinding.ActivityViewCanvasBinding;
import com.hjc.baselib.activity.BaseActivity;
import com.hjc.baselib.viewmodel.CommonViewModel;

/**
 * @Author: HJC
 * @Date: 2019/10/24 14:56
 * @Description: 画布操作
 */
@Route(path = RoutePath.URL_VIEW_CANVAS)
public class ViewCanvasActivity extends BaseActivity<ActivityViewCanvasBinding, CommonViewModel> {


    @Override
    public int getLayoutId() {
        return R.layout.activity_view_canvas;
    }

    @Override
    protected CommonViewModel getViewModel() {
        return null;
    }

    @Override
    public void initData(Bundle savedInstanceState) {

    }

    @Override
    public void addListeners() {
        mBindingView.setOnClickListener(this);

        mBindingView.titleBar.setOnViewLeftClickListener(view -> finish());
    }

    @SuppressLint("NonConstantResourceId")
    @Override
    public void onSingleClick(View v) {
        switch (v.getId()) {
            case R.id.btn1:
                mBindingView.customView.draw(1);
                break;

            case R.id.btn2:
                mBindingView.customView.draw(2);
                break;

            case R.id.btn3:
                mBindingView.customView.draw(3);
                break;

            default:
                break;
        }
    }
}

File: module-senior/src/main/java/com/hjc/module_senior/ui/view/ViewCanvasActivity.kt
package com.hjc.module_senior.ui.view

import android.os.Bundle
import android.view.View
import com.alibaba.android.arouter.facade.annotation.Route
import com.gyf.immersionbar.ImmersionBar
import com.hjc.library_base.activity.BaseActivity
import com.hjc.library_base.viewmodel.CommonViewModel
import com.hjc.library_common.router.RoutePath
import com.hjc.library_widget.bar.OnViewLeftClickListener
import com.hjc.module_senior.R
import com.hjc.module_senior.databinding.SeniorActivityViewCanvasBinding

/**
 * @Author: HJC
 * @Date: 2019/10/24 14:56
 * @Description: 画布操作
 */
@Route(path = RoutePath.Senior.URL_VIEW_CANVAS)
class ViewCanvasActivity : BaseActivity<SeniorActivityViewCanvasBinding, CommonViewModel>() {

    override fun getLayoutId(): Int {
        return R.layout.senior_activity_view_canvas
    }

    override fun createViewModel(): CommonViewModel? {
        return null
    }

    override fun getImmersionBar(): ImmersionBar? {
        return ImmersionBar.with(this)
            .fitsSystemWindows(true)
            .statusBarColor(R.color.senior_color)
    }

    override fun initData(savedInstanceState: Bundle?) {

    }

    override fun addListeners() {
        mBindingView.onClickListener = this

        mBindingView.titleBar.setOnViewLeftClickListener(object : OnViewLeftClickListener {
            override fun leftClick(view: View?) {
                finish()
            }
        })
    }

    override fun onSingleClick(v: View?) {
        when (v?.id) {
            R.id.btn1 -> mBindingView.customView.draw(1)

            R.id.btn2 -> mBindingView.customView.draw(2)

            R.id.btn3 -> mBindingView.customView.draw(3)

            else -> {
            }
        }
    }
}